dotnet/src/InternalUtilities/src/Data/VectorStoreRecordPropertyReader.cs

```diff
@@ -254,6 +254,23 @@ public static VectorStoreRecordDefinition CreateVectorStoreRecordDefinitionFromT
     /// <param name="supportEnumerable">A value indicating whether <see cref="IEnumerable{T}"/> versions of all the types should also be supported.</param>
     /// <exception cref="ArgumentException">Thrown if any of the properties are not in the given set of types.</exception>
     public static void VerifyPropertyTypes(List<PropertyInfo> properties, HashSet<Type> supportedTypes, string propertyCategoryDescription, bool? supportEnumerable = false)
+    {
+        var supportedEnumerableTypes = supportEnumerable == true
+            ? supportedTypes
+            : [];
+
+        VerifyPropertyTypes(properties, supportedTypes, propertyCategoryDescription, supportedEnumerableTypes);
+    }
+
+    /// <summary>
+    /// Verify that the given properties are of the supported types.
+    /// </summary>
+    /// <param name="properties">The properties to check.</param>
+    /// <param name="supportedTypes">A set of supported types that the provided properties may have.</param>
+    /// <param name="propertyCategoryDescription">A description of the category of properties being checked. Used for error messaging.</param>
+    /// <param name="supportedEnumerableTypes">A set of supported types that the provided enumerable properties may use as their element type.</param>
+    /// <exception cref="ArgumentException">Thrown if any of the properties are not in the given set of types.</exception>
+    public static void VerifyPropertyTypes(List<PropertyInfo> properties, HashSet<Type> supportedTypes, string propertyCategoryDescription, HashSet<Type> supportedEnumerableTypes)
```

Comment:
@westey-m refactoring to support cases where not all types are supported as element types for enumerables

Reply:
Thanks, looks good.